### PR TITLE
Feature/add pupil page

### LINKED ISSUE #2: AddPupilEvaluation Page
## Value proposition
**As a** user
**I need** a way to add a pupil and a evalution
**to be able to** see all evaluations of a pupil at one place.

## Acceptance criteria
- [x] All inputs are required fields.
- [x] If the pupil already exists, the evaluation should be added to its card at the bottom (instead of creating a new card).
- [x] The list (cards of pupils) should scroll vertically on overflow.
- [x] The DeleteButton should delete the card.

## Tasks

- [x] Form component
- [x] Card component
- [x] Integrate components into App
- [x] Setup deployment

### LINKED ISSUE #12: Back Button for addPupil form
There is a back button missing in the  user story of sprint 1:
wihtout the button, the user can only add a pupil when encountering the form, but cannot leave the form without adding.

### PR DESCRIPTION
**Pull request by PReichetanz**
Bringing together all components and add styling:

- Rename MainButton to Button for better handling
- Make Form display dependent on state via buttons
- Add container with grid layout
- Add header
- Add backButton to Form (closes #12, but was not part of the user story) 

<div>

</div>

[View on heroku](https://capstone-rev-feature-ad-b834ot.herokuapp.com/)
[View Storybook](https://capstone-rev-feature-ad-b834ot.herokuapp.com/storybook/?path=/story/component-button--regular)

closes #2

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 type Pupil = {
   name: string;
   evaluations: string[];
 };
 
-function App(): JSX.Element {
+export default function App(): JSX.Element {
   const [pupils, setPupils] = useState<Pupil[]>([]);
-  console.log(pupils);
+  const [displayState, setDisplayState] = useState('');
```
Comment (lajanzen) [inline src/app/App.tsx]: It seems like you only have 2 states here, so I would rather work with true/false: 

```suggestion
  const [isFormShown, setIsFormShown] = useState(false);
```

File: src/app/App.tsx [hunk 2]
```diff
@@ -36,6 +38,7 @@ function App(): JSX.Element {
           evaluations: [pupil.evaluation],
         },
       ]);
+      setDisplayState('');
```
Comment (lajanzen) [inline src/app/App.tsx]:
```suggestion
      setIsFormShown(false);
```

File: src/app/App.tsx [hunk 3]
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
...
-          deleteCard={deletePupil}
-        />
-      ))}
-    </>
+    <Container>
+      <Header>
+        <h1>Meine Klasse</h1>
+      </Header>
+      <Main>
+        {displayState === 'showAddPupilForm' && (
```
Comment (lajanzen) [inline src/app/App.tsx]:
```suggestion
        {isFormShown && (
```

File: src/app/App.tsx [hunk 4]
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
...
+          <Form
+            nameLabel="Name des Schülers"
+            evaluationLabel="Worturteil"
+            onSubmit={handleFormSubmit}
+            submitted={false}
+            onCancel={() => setDisplayState('')}
```
Comment (lajanzen) [inline src/app/App.tsx]:
```suggestion
            onCancel={() => setIsFormShown(false)}
```

File: src/app/App.tsx [hunk 5]
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
...
+        )}
+        {pupils.map((pupil, key) => (
+          <Card
+            pupil={pupil}
+            key={`${pupil.name}-${key}`}
+            deleteCard={deletePupil}
+          />
+        ))}
+      </Main>
+      <AddButton onClick={() => setDisplayState('showAddPupilForm')}>
```
Comment (lajanzen) [inline src/app/App.tsx]:
```suggestion
      <AddButton onClick={() => setIsFormShown(true)}>
```

File: src/app/App.tsx [hunk 6]
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
...
+  color: var(--color-text-white);
+  background: var(--color-background-dark);
+  padding-left: 1rem;
+  border-bottom: 1px solid var(--color-stroke);
+`;
+
+const Container = styled.div`
+  height: 100vh;
+  display: grid;
+  grid-template-columns: 1fr;
```
Comment (lajanzen) [inline src/app/App.tsx]: Do you need this?
```suggestion
```

File: src/app/components/Button/Button.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import Button from './Button';
+
+export default {
+  title: 'Component/Button',
+  component: Button,
+};
+
+export const Regular = (): JSX.Element => (
+  <Button onClick={() => console.log('clicked!')}>Schüler hinzufügen</Button>
```
Comment (derjubu) [inline src/app/components/Button/Button.stories.tsx:10]: I think you can remove this onClick function, if it only logs something in the console for control purposes during development.

**Review by derjubu — APPROVED** [review]

File: src/app/components/Form/Form.tsx [hunk 8]
```diff
@@ -32,44 +35,66 @@ export default function Form({
...
+          }}
+        />
+        <label htmlFor="name">{nameLabel}:</label>
+        <Input
+          type="text"
+          id="name"
+          placeholder="Lena Beispiel"
+          onChange={(event) => setName(event.target.value)}
+          value={name}
+          submitted={inputError}
```
Comment (lajanzen) [inline src/app/components/Form/Form.tsx]: What does this do exactly?

File: src/app/components/Button/Button.stories.tsx (hunk 7 above)
Comment (PReichetanz) [reply src/app/components/Button/Button.stories.tsx:10]: I really would like to, but TypeScript needs it to be happy!

File: src/app/App.tsx (hunk 6 above)
Comment (PReichetanz) [reply src/app/App.tsx]: I don't, thanks!

File: src/app/components/Form/Form.tsx (hunk 8 above)
Comment (PReichetanz) [reply src/app/components/Form/Form.tsx]: It checks whether an input is missing after submitting. I renamed it to "missingInput", so it should be more descriptive.

**Pull request merged**